typings/index.d.ts

```diff
@@ -733,6 +733,98 @@ declare module "noblox.js" {
         price?: number;
     }
 
+    
+    type AvatarType = "MorphToR6" | "MorphToR15" | "PlayerChoice"
+    type AnimationType = "Standard" | "PlayerChoice"
+    type CollisionType = "InnerBox" | "OuterBox"
+    type JointType = "Standard" | "ArtistIntent"
+
+    type Genre = "All" | "Tutorial" | "Scary" | "TownAndCity" | "War" | "Funny" | "Fantasy" | "Adventure" | "SciFi" | "Pirate" | "FPS" | "RPG" | "Sports" | "Ninja" | "WildWest"
+    type PlayableDevices = "Computer" | "Phone" | "Tablet" | "Console"
+    type Regions = "Unknown" | "China"
+
+    interface UniverseAsset 
+    {
+        assetID: number,
+        assetTypeID: number,
+        isPlayerChoice: boolean
+    }
+
+    interface UniversePermissions
+    {
+        IsThirdPartyTeleportAllowed?: boolean;
+        IsThirdPartyAssetAllowed?: boolean;
+        IsThirdPartyPurchaseAllowed?: boolean;
+    }
+
+    interface UniverseSettings
+    {
+        allowPrivateServers?: boolean;
+        privateServerPrice?: number;
+
+        name?: string;
+        description?: string;
+
+        universeAvatarType?: AvatarType;
+        universeAnimationType?: AnimationType;
+        universeCollisionType?: CollisionType;
+        universeJointPositioningType?: JointType;
+        
+        isArchived?: boolean;
+        isFriendsOnly?: boolean;
+
+        genre?: Genre;
+
+        playableDevices?: Array<PlayableDevices>;
+        universeAvatarAssetOverrides?: Array<UniverseAsset>;
+
+        isForSale?: boolean;
+        price?: number;
+        
+        universeAvatarMinScales?: AvatarScale
+        universeAvatarMaxScales?: AvatarScale
+
+        studioAccessToApisAllowed?: boolean;
+        permissions?: UniversePermissions;
+
+        optInRegions?: Array<Regions>;
+        optOutRegions?: Array<Regions>;
+    }
+
+    interface UpdateUniverseResponse
```

Comment:
Could this interface be re-written as an [subtype](https://www.typescriptlang.org/docs/handbook/interfaces.html#extending-interfaces) of UniverseSettings. Unless I have missed anything it is identical except it also has 'id'.

Reply:
I note optOutRegions is not present. I'm also not seeing that in the [docs](https://develop.roblox.com/docs#!/UniverseSettings/patch_v1_universes_universeId_configuration)? Is that an undocumented value?

Reply:
This data is compiled from the data used to create the docs page. The model for this endpoint included it, so I added it in case they decide to use it in the future.

Reply:
I knew there was something it was forgetting and it was subtypes 😂 I'll have to experiment with better docs to see how it reacts to a subtype in its typings file.